### PR TITLE
NDRS-1102: Secret key should not implement clone

### PR DESCRIPTION
**Pull request by darthsiroftardis**
REF: https://casperlabs.atlassian.net/browse/NDRS-1102

CHANGELOG:

- Removed `Clone` trait from Asymetric Secret Key type
- Refactoring to account for above change

### CODE REVIEW

File: node/src/utils/external.rs [hunk 1]
```diff
@@ -22,6 +22,7 @@ use casper_types::SecretKey;
 
 use super::{read_file, ReadFileError};
 use crate::{crypto, crypto::AsymmetricKeyExt, tls};
+use std::sync::Arc;
```
Comment (Fraser999) [inline node/src/utils/external.rs:25]: This should be moved.

File: types/src/crypto/asymmetric_key/gens.rs [hunk 2]
```diff
@@ -10,6 +10,7 @@ use proptest::{
 
 use crate::{crypto::SecretKey, PublicKey};
 
+/*
```
Comment (Fraser999) [inline types/src/crypto/asymmetric_key/gens.rs:13]: I guess this code block should be removed?

**Review by xcthulhu — APPROVED** [review]
Aside from the points @Fraser999 identified this looks good to me.

File: node/src/components/consensus/config.rs [hunk 3]
```diff
@@ -9,14 +9,15 @@ use crate::{
     types::{chainspec::HighwayConfig as HighwayProtocolConfig, Chainspec, TimeDiff, Timestamp},
     utils::External,
 };
+use std::sync::Arc;
```
[the remark this answers is not in the document]
Comment (darthsiroftardis) [reply node/src/components/consensus/config.rs]: Moved

File: node/src/reactor/validator/tests.rs [hunk 4]
```diff
@@ -199,14 +202,16 @@ async fn run_equivocator_network() {
 
     let mut rng = crate::new_rng();
 
-    let alice_sk = SecretKey::random(&mut rng);
+    let alice_sk = Arc::new(SecretKey::ed25519_from_bytes([0; SecretKey::ED25519_LENGTH]).unwrap());
```
[the remark this answers is not in the document]
Comment (darthsiroftardis) [reply node/src/reactor/validator/tests.rs]: Changed back to random, I must've done it mindlessly as part of the refactor

File: node/src/utils/external.rs (hunk 1 above)
Comment (darthsiroftardis) [reply node/src/utils/external.rs]: Moved

File: types/src/crypto/asymmetric_key/gens.rs (hunk 2 above)
Comment (darthsiroftardis) [reply types/src/crypto/asymmetric_key/gens.rs]: Code block removed

**Review by Fraser999 — APPROVED** [review]
LGTM

**Comment by darthsiroftardis** [comment]
bors merge

**Pull request merged**